include/fmt/format.h

```diff
@@ -3305,6 +3306,66 @@ arg_join<internal::iterator_t<const Range>, wchar_t> join(const Range& range,
   return join(std::begin(range), std::end(range), sep);
 }
 
+template <typename Char, typename... T> struct tuple_arg_join : internal::view {
```

Comment:
Let's move all of this to `ranges.h`.

Reply:
For some reason, the test suite wasn't finding the symbols I was defining there in the ranges-test.cc file. Was I just doing something silly? I can try again later.